src/watcher.js

```diff
@@ -0,0 +1,38 @@
+/* @flow */
+import Watchpack from 'watchpack';
+import debounce from 'debounce';
+
+import {createLogger} from './util/logger';
+
+const log = createLogger(__filename);
+
+
+export default function onSourceChange(
+    {sourceDir, artifactsDir, onChange}: Object): Watchpack {
+  // TODO: For network disks, we would need to add {poll: true}.
+  const watcher = new Watchpack();
+
+  const executeImmediately = true;
+  watcher.on('change', debounce((filePath) => {
+    proxyFileChanges({artifactsDir, onChange, filePath});
+  }, 1000, executeImmediately));
```

Comment:
I think that is not important or mandatory, but it probably worth a mention:
I'm wondering if in some scenarios it could be helpful being able to configure the debounce time (e.g. in case the changes are generated from another build tool, like webpack or browserify).